conformance/lang/expressions/range-expr/range_expr_exclusive.balt

```diff
@@ -0,0 +1,5877 @@
+
+    foreach int i in 1*2 ..< 36/4 {
+          arr.push(i); 
+          count = count + 1;
+    }
+    io:println(count); // @output 7
+    io:println(arr); // @output [2,3,4,5,6,7,8]
+
+    count = 0;
+    arr = [];
+    foreach int i in 34%5 ..< 4*5 {
+          arr.push(i); 
+          count = count + 1;
+    }
+    io:println(count); // @output 16
+    io:println(arr); // @output [4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19]
+}
+
+Test-Case: output
+Description: Test binary bitwise expressions as expressions in range expression (exclusive).
+Labels: range-expr, iterable-exclusive, binary-bitwise-expr, DecimalNumber, array-type, list-constructor-expr,
+        array:push, int, additive-expr
+
+function init() {
+    int count = 0;
+    int[] arr = [];
+
+    foreach int i in (1&2) ..< (4|8) {
+          arr.push(i); 
+          count = count + 1;
+    }
+    io:println(count); // @output 12
+    io:println(arr); // @output [0,1,2,3,4,5,6,7,8,9,10,11]
+
+    count = 0;
+    arr = [];
+    foreach int i in (6|3) ..< (2^8) {
+          arr.push(i); 
+          count = count + 1;
+    }
+    io:println(count); // @output 3
+    io:println(arr); // @output [7,8,9]
+}
+
+Test-Case: output
+Description: Test checking expressions as expressions in range expression (exclusive).
+Labels: range-expr, iterable-exclusive, check, checkpanic, DecimalNumber, error, array-type, list-constructor-expr,
+        array:push, int, union-type, optional-type, additive-expr
+
+function init() returns error? {
+    int|error startIndex = 2;
+    int|error endIndex = 12;
+    int count = 0;
+    int[] arr = [];
+
+    foreach int i in check startIndex ..< checkpanic endIndex {
+          arr.push(i); 
+          count = count + 1;
+    }
+    io:println(count); // @output 10
+    io:println(arr); // @output [2,3,4,5,6,7,8,9,10,11]
+}
+
+Test-Case: output
+Description: Test let expression as expressions in range expression (exclusive).
+Labels: range-expr, iterable-exclusive, let-expr, DecimalNumber, int
+
+function init() {
+    int count = 0;
+    int[] arr = [];
+
+    foreach int i in (let int h1 = 10 in h1) ..< (let int h1 = 25 in h1) {
+          arr.push(i); 
+          count = count + 1;
+    }
+    io:println(count); // @output 15
+    io:println(arr); // @output [10,11,12,13,14,15,16,17,18,19,20,21,22,23,24]
+}
+
+Test-Case: output
+Description: Test shift expressions as expressions in range expression (exclusive).
+Labels: range-expr, iterable-exclusive, shift-expr, DecimalNumber, int
+
+function init() {
+    int count = 0;
+    int[] arr = [];
+
+    foreach int i in 1 << 2 ..< 100 >> 4 {
+          arr.push(i); 
+          count = count + 1;
+    }
+    io:println(count); // @output 2
+    io:println(arr); // @output [4,5]
+}
+
+Test-Case: output
+Description: Test conditional expressions as expressions in range expression (exclusive).
+Labels: range-expr, iterable-exclusive, conditional-expr, DecimalNumber, error, array-type, list-constructor-expr,
```

Comment:
we decided to use `ternary-conditional-expr` for ternary and `binary-conditional-expr` for elvis  instead of `conditional-expr`